### PR TITLE
DDS-1209 [Card] Add Card Component

### PR DESCRIPTION
**Pull request by rinjInTokyo**
This PR adds Card (`daikin-card`) component.

## Related Ticket(s) and Links

- JIRA Ticket: [DDS-1209](https://daikinsvoil.atlassian.net/browse/DDS-1209)
- Figma design: [Figma](https://www.figma.com/design/VyaaU8Ta9yzyf0PsURWSSf/DDS%3A-Design-Kit?node-id=3406-2962&node-type=canvas&m=dev)

## API Design

- [JIRA Wiki (card)](https://daikinsvoil.atlassian.net/wiki/spaces/GDS/pages/3103522843/Card+Component+API)
- [JIRA Wiki (card header)](https://daikinsvoil.atlassian.net/wiki/spaces/GDS/pages/3399778419/Card+Header+Component+API)
- [JIRA Wiki (card footer)](https://daikinsvoil.atlassian.net/wiki/spaces/GDS/pages/3103555618/Card+Footer+Component+API)

### Visual Regression Tests

- Card
    -  \[outline; ghost;\] with border and \[body; footer; both\] structure.
- Card Header
    -  \[icon; description; both;\] structure
- Card Footer
    -  \[button; link;\] the action element in footer

### Interaction Tests

- Because it's a wrap component so not interaction now

## Checklist

- [x] I have added a changeset file that describes the changes.  
       _Hint_: run `npx changeset`.
- [x] I have performed a self-review of my code.
- [x] I have added appropriate visual regression tests.
- [x] I have added appropriate interaction tests.
- [x] I have confirmed that there are no accessibility warnings that can be addressed.  
       _Hint_: check the Accessibility tab in the Storybook.
- [x] I have confirmed that the component can be operated using only the keyboard.
- [x] I have confirmed that the focus state of the component is visually discernible.

[DDS-1209]: https://daikinsvoil.atlassian.net/browse/DDS-1209?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/card-header/daikin-card-header.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { css, html, LitElement, unsafeCSS } from "lit";
+import { customElement } from "lit/decorators.js";
+
+import { cva } from "class-variance-authority";
+import tailwindStyles from "../../tailwind.css?inline";
+
+const cvaTitle = cva([
+  "text-ddt-color-common-text-primary",
```
Comment (yodas7) [inline src/components/card-header/daikin-card-header.ts:8]:
```suggestion
  "flex-none",
  "text-ddt-color-common-text-primary",
```

File: src/components/card-header/daikin-card-header.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  "font-daikinSerif",
+  "text-base",
+  "leading-[1.3]",
+  "not-italic",
+  "font-bold",
+]);
```
Comment (yodas7) [inline src/components/card-header/daikin-card-header.ts:14]:
```suggestion
])();
```

File: src/components/card-header/daikin-card-header.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+const cvaDescription = cva([
+  "text-ddt-color-common-text-secondary",
```
Comment (yodas7) [inline src/components/card-header/daikin-card-header.ts:17]:
```suggestion
  "flex-none",
  "text-ddt-color-common-text-secondary",
```

File: src/components/card-header/daikin-card-header.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  "font-daikinSerif",
+  "text-sm",
+  "leading-normal",
+  "not-italic",
+  "font-normal",
+]);
```
Comment (yodas7) [inline src/components/card-header/daikin-card-header.ts:23]:
```suggestion
])();
```

File: src/components/card/daikin-card.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import tailwindStyles from "../../tailwind.css?inline";
+
+const cvaContainer = cva(
+  [
+    "flex",
+    "rounded-lg",
+    "bg-ddt-color-common-background-default",
+    "p-4",
+    "gap-4",
+    "flex-col",
```
Comment (yodas7) [inline src/components/card/daikin-card.ts:14]: nit.

```suggestion
    "flex",
    "flex-col",
    "gap-4",
    "p-4",
    "rounded-lg",
    "bg-ddt-color-common-background-default",
```

**Review by yodas7 — COMMENTED** [review]
✅ HTML tags & ARIA roles (Radix UI, MUI)
✅ Component Sizable

File: src/components/card-header/stories/framework-react.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+});
+
+export const metadata: Meta<DaikinCardHeaderStoryArgs> = {
+  component: ({
+    label,
+    description,
+    leftIcon,
+    ...props
+  }: DaikinCardHeaderStoryArgs) => (
+    <div style={{ width: 248 }}>
```
[the remark this answers is not in the document]
Comment (rinjInTokyo) [reply src/components/card-header/stories/framework-react.tsx:22]: @yodas7 
> an unit is needed (px?)

Miss that

> not sure why 248.

The header's width spans the entire available space, so I try to set it to match the width specified in the figma design to look better and improve the accuracy of visual testing.

File: src/components/card-header/daikin-card-header.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+    :host {
+      display: flex;
+      align-items: center;
+    }
+  `;
+
+  override render() {
+    return html` <div class="flex w-full gap-1">
+      <slot name="left-icon" class="text-ddt-color-common-text-primary"></slot>
```
[the remark this answers is not in the document]
Comment (rinjInTokyo) [reply src/components/card-header/daikin-card-header.ts:59]: It seems `size-6 icon-size-full` will always hold a 24px space even there is no icon. So I refer to button component to use `icon-size-6`

File: src/components/card-header/daikin-card-header.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+ * @slot - A slot for card title.
+ *
+ * @example
+ *
+ * ```html
+ * <!-- See `daikin-card` component for complete example. -->
+ * <daikin-card-header>
+ *   <daikin-icon slot="left-icon" icon="alarm" size="xl" color="current"></daikin-icon>
+ *   <span slot="description">Description</span>
+ *   <span>Label Title</span>
```
Comment (yodas7) [inline src/components/card-header/daikin-card-header.ts:43]: nit.

```suggestion
 *   <span>Card Title</span>
```

File: src/components/card-footer/daikin-card-footer.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    :host {
+      display: flex;
+      align-items: center;
+      width: 100%;
+      justify-content: flex-end;
+    }
+  `;
+
+  override render() {
+    return html` <slot></slot> `;
```
Comment (yodas7) [inline src/components/card-footer/daikin-card-footer.ts:38]:
```suggestion
    return html`<slot></slot>`;
```

File: src/components/card-header/daikin-card-header.visual.test.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+type StoryArgs = InferStorybookArgTypes<typeof DAIKIN_CARD_HEADER_ARG_TYPES>;
+
+const getPageURL = (args: StoryArgs = {}) =>
+  getStorybookIframeURL("components-card-header--default", args);
+
+describeEach(["light", "dark"] as const, (theme) => {
+  describeEach(["icon", "description", "both"], (structure) => {
+    const baseArgs = {
+      $theme: theme,
+      label: "Label Title",
```
Comment (yodas7) [inline src/components/card-header/daikin-card-header.visual.test.ts:19]: nit.

```suggestion
      label: "Card Title",
```

File: src/components/card/stories/framework-react.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  component: ({ ...props }: DaikinCardStoryArgs) => (
+    <ReactDaikinCard {...props}>
+      <ReactDaikinCardHeader>
+        <ReactDaikinIcon
+          slot="left-icon"
+          icon="alarm"
+          size="current"
+          color="current"
+        />
+        <span>Label Title</span>
```
Comment (yodas7) [inline src/components/card/stories/framework-react.tsx:28]: nit.

```suggestion
        <span>Card Title</span>
```

File: src/components/card/stories/framework-wc.ts [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+export const metadata: Meta<DaikinCardStoryArgs> = {
+  title: "Components/Card",
+  tags: ["autodocs"],
+  render: (args) => html`
+    <daikin-card .outline=${args.outline}>
+      <daikin-card-header>
+        <daikin-icon slot="left-icon" icon="alarm" size="xl" color="current">
+        </daikin-icon>
+        <span>Label Title</span>
```
Comment (yodas7) [inline src/components/card/stories/framework-wc.ts:18]:
```suggestion
        <span>Card Title</span>
```

File: .changeset/calm-apes-allow.md [hunk 13]
```diff
@@ -0,0 +1,5 @@
+---
+"@daikin-oss/design-system-web-components": patch
+---
+
+Add Card Component. (DDS-1209)
```
Comment (yodas7) [inline .changeset/calm-apes-allow.md:5]: We now treat new component as minor update, though it does not include breaking changes.

```suggestion
---
"@daikin-oss/design-system-web-components": minor
---

Card: Initial Implementation. (DDS-1209)
```

File: src/components/card-footer/index.ts [hunk 14]
```diff
@@ -0,0 +1 @@
+export * from ".//daikin-card-footer";
```
Comment (yodas7) [inline src/components/card-footer/index.ts:1]:
```suggestion
export * from "./daikin-card-footer";
```

**Review by yodas7 — CHANGES_REQUESTED** [review]

**Review by yodas7 — APPROVED** [review]
LGTM

**Review by hiro-daikin — APPROVED** [review]

**Pull request merged**